### ISSUE TITLE
Descriptor schema (or just actually be JSON Schema)

### ISSUE THREAD

**Issue by jmakeig** [marklogic/entity-services#311]
With a schema, many editors will provide type-ahead hints when editing JSON, for example, [in VSCode](https://code.visualstudio.com/docs/languages/json#_mapping-in-the-user-settings). We should provide a JSON Schema for our Entity Type Descriptor. We should also publish this at a URL so that it can be referenced in config, such as in the example above.

Longer term, we should think about actually being JSON Schema, rather than kinda like JSON Schema.

**Comment by grechaw** [comment]
You realize that this request is for a descriptor for the es in a different version of JSON schema than es was actually based on?  I was pretty surprised how different the new JSON schema is from what we used for entity services.

Also, the different between entity services and JSON schema is about supporting datatypes.  I've not seen any JSON schema that supports datatypes reasonably.

So, laudable goal, but as you've assigned this to yourself, it looks like you recognize we need a little strategy as well as a task.

**Comment by jmakeig** [comment]
Yes, mostly place-holder for now. Title should be “Make a kick-ass editor experience for Entity Services model descriptors”.

**Comment by pdesmarets** [comment]
Take a look at https://hackolade.com/nosqldb.html#marklogic  A visual JSON Schema editor build for NoSQL databases.  Still need to do reverse-engineering of MarkLogic, but almost there...

**Comment by grechaw** [comment]
Thanks @pdesmarets , I'll look forward to that.  I've tinkered slightly with hackolade and think it would be a great fit for us too.

**Comment by pdesmarets** [comment]
We added a ML plugin a few months.  Would love to get your feedback for future enhancements.
